src/ast/stmt.rs

```diff
@@ -74,8 +76,8 @@ pub struct IfStmt {
 
 #[derive(Debug, Eq, PartialEq, Serialize, Clone, new)]
 pub struct ForStmt {
-    // ExprStmt(BinExpr) or DeclStmt
-    pub init: Option<Box<Stmt>>,
+    // Containing ExprStmt(BinExpr) or DeclStmt commonly
+    pub init: Vec<Stmt>,
     pub condition: Option<Expr>,
     pub post: Option<Expr>,
```

Comment:
Since we're updating the `init` block, we should probably also update the post block for the same reason (since you can have `for (...; ...;i++, j++) { ... }`).

Reply:
good call. will do that right now